File: HW01_407235049.py
import math


def main():
    print("Please Input number:", end='')
    
    
    #use map function to input with the spit() function 
   
    a, b, c = map(float, input().split())
    rule_triangle = rule(a, b, c)
    time = 0

    #use open to open the file

    file = open("python_result.txt", mode='a')
    while rule_triangle != 1:
        time = time + 1
        print("Question "+str(time)+" Input are a="+str(a)+", b="+str(b)+", c="+str(c))
        file.write("Question "+str(time)+" Input are a="+str(a)+", b="+str(b)+", c="+str(c)+"\n")
        if rule_triangle == 2:
            print("The Input is not legal.")
            file.write("The Input is not legal.\n")
            print("The Input numbers must be positive.")
            print("Please Input number:", end='')
            a, b, c = map(float, input().split())
            rule_triangle = rule(a, b, c)
            continue
        elif rule_triangle == 3:
            print("The Input is not legal.")
            file.write("The Input is not legal.\n")
            print("The total of two of the Input numbers must"
                  "be greater than another Input number.")
            print("Please Input number:", end='')
            a, b, c = map(float, input().split())
            rule_triangle = rule(a, b, c)
            continue
        s = (float(a + b + c))/2
        area = math.sqrt(s * (s-a) * (s-b) * (s-c))
        print("The area of the triangle is : ", format(area, '.4f'))
        file.write("The area of the triangle is :"+str(format(area, '.4f'))+"\n")
        print("Please Input number:", end='')
        a, b, c = map(float, input().split())
        rule_triangle = rule(a, b, c)
    print("Program exit.")
    file.write("Program exit.\n")
    file.close()
    return


def rule(a, b, c):

    #each of the number to return means the different of the situation
    
    if a == -1 and b == -1 and c == -1:
        return 1
    elif a+b <= c or a+c <= b or b+c <= a or a < 0 or b < 0 or c < 0:
        if a < 0 or b < 0 or c < 0:
            return 2
        else:
            return 3
    else:
        return 0


main()

File: HW01_407235049.java
import java.util.Scanner;
import java.io.*;
public class HW01_407235049
{
    public static int rule(double a, double b, double c)
    {
         /**
         * each of the parameter represent each of the triangle of the edge
         * @param a
         * @param b
         * @param c
         * @return 1 means program exit
         * @return 2 means one of the input is not greater than 0
         * @return 3 means triangle edge that did not follow the rule
         * @return 0 means that it follow the rule
         */
        if(a==-1&&b==-1&&c==-1)
            return 1;
        else if(a+b<=c||a+c<=b||b+c<=a||a<0||b<0||c<0)
        {
            if(a<0||b<0||c<0)
                return 2 ;
            else
                return 3 ;
        }
        else
            return 0;
    }
    public static void main(String[] args) throws IOException
    {
        FileWriter Fw = new FileWriter("java_result.txt");
        Scanner scan = new Scanner(System.in);
        double a, b, c, s, area;
        System.out.print("Please Input first number:");
        a = scan.nextDouble();
        b = scan.nextDouble();
        c = scan.nextDouble();
        int rule_triangle = rule(a, b, c);
        int time = 0;
        String illegal = "The Input is not legal.\n";
        while (rule_triangle != 1)
        {
            time++;
            System.out.println("Question "+time+" Inputs are a="+a+", b="+b+
                    ", c="+c);
            String inputTime = "Question "+time+" Inputs are a="+a+", b="+b+
                    ", c="+c+"\n";
            Fw.write(inputTime);
            if (rule_triangle == 2)
            {
                System.out.println("The Input is not legal.");
                Fw.write(illegal);
                System.out.println("The Input numbers must be positive.");
                System.out.print("Please Input first number:");
                a = scan.nextDouble();
                b = scan.nextDouble();
                c = scan.nextDouble();
                rule_triangle = rule(a, b, c);
                continue ;
            }
            else if (rule_triangle == 3)
            {
                System.out.println("The Input is not legal.");
                Fw.write(illegal);
                System.out.println("The total of two of the Input numbers must " +
                        "be greater than another Input number.");
                System.out.print("Please Input first number:");
                a = scan.nextDouble();
                b = scan.nextDouble();
                c = scan.nextDouble();
                rule_triangle = rule(a, b, c);
                continue;
            }
            s = (a + b + c) / 2;
            area = Math.sqrt(s * (s - a) * (s - b) * (s - c));
            System.out.println("The area of the triangle is : "+(double)Math.round(area*10000)/10000);
            String output = "The area of the triangle is :" +(double)Math.round(area*10000)/10000 + "\n" ;
            Fw.write(output);
            System.out.print("Please Input first number:");
            a = scan.nextDouble();
            b = scan.nextDouble();
            c = scan.nextDouble();
            rule_triangle = rule(a, b, c);
        }
        System.out.println("Program exit.");
        String exit = "Program exit.\n";
        Fw.write(exit);
        Fw.close();
    }
}
